test/info.py

```diff
@@ -62,6 +62,14 @@
     'lpc11u35_archlink_if': 0x9013,
 }
 
+# Maps a target application filename (sans extension) to the ID of the board 
+# it is built to run on. This mapping is also used to display a board name 
+# in the test output. Thus, name the target applications after the  board to
+# avoid confusing results.
+#
+# One-to-many is fine. Many-to-one is fine as long as only one (or none) target 
+# application is found on disk per connected board ID
+#
```

Comment:
Currently all the entries in TARGET_NAME_TO_BOARD_ID correspond to the boards on https://developer.mbed.org/platforms/<board name here>.  The compile API used for testing is based on the URL of the platform page for the board.  For example, the 'NXP-LPC800-MAX' page can be found here https://developer.mbed.org/platforms/NXP-LPC800-MAX/.